doc/Paypal_Payments.md

```diff
@@ -0,0 +1,147 @@
+# Documentation of PayPal payments
+
+## Steps of a successful donation payment
+
+1. The user submits the donation form
+2. The Fundraising Application stores donation data in the database,
+   generating an ID for the donation. The payment state of the donation is
+   "incomplete"
+3. The Fundraising Application redirects the user to PayPal page, with the
+   donation ID, access token, confirmation and notification routes (see
+   below) as extra URL parameters
+4. The User submits the PayPal form
+5. PayPal redirects the user to success page of the Fundraising
+   Application (passed to PayPal in step 3), passing back the donation ID
+   and access token as URL parameters
+6. PayPal sends an HTTP request called **"IPN"** ([Instant Payment
+   Notification][1]
+   to the "PayPal Notification" route (passed to PayPal in step 3) of the
+   Fundraising Application. The HTTP request contains payment metadata
+   (amount, currency, date, account information) and the donation ID.
+7. The Fundraising Application verifies the received data with PayPal (via
+   HTTP)
+8. The Fundraising Application validates the data and marks the payment
+   state donation as "booked".
+
+
+## Metadata for PayPal
+
+When redirecting the user to PayPal, we need to specify the following
+data:
+
```

Comment:
Should we mark the field that's used to link the IPN back to the donation/membership request?

Reply:
That would be donation ID. Yes, I'll update the document